engine/src/main/java/org/terasology/rendering/assets/skeletalmesh/SkeletalMeshData.java

```diff
@@ -74,62 +73,57 @@ public Bone getRootBone() {
     }
 
     public List<Vector3f> getBindPoseVertexPositions() {
-        List<Vector3f> positions = Lists.newArrayListWithCapacity(bones.size());
-        List<Quat4f> rotations = Lists.newArrayListWithCapacity(getBones().size());
+        Matrix4f[] transforms = new Matrix4f[bones.size()];
         for (Bone bone : bones) {
-            positions.add(bone.getObjectPosition());
-            rotations.add(bone.getObjectRotation());
+            transforms[bone.getIndex()] = bone.getObjectTransform();
         }
-        return getVertexPositions(positions, rotations);
+        return getVertexPositions(Arrays.asList(transforms));
     }
 
     public List<Vector3f> getBindPoseVertexNormals() {
-        List<Vector3f> positions = Lists.newArrayListWithCapacity(bones.size());
-        List<Quat4f> rotations = Lists.newArrayListWithCapacity(getBones().size());
+        Matrix4f[] transforms = new Matrix4f[bones.size()];
         for (Bone bone : bones) {
-            positions.add(bone.getObjectPosition());
-            rotations.add(bone.getObjectRotation());
+            transforms[bone.getIndex()] = bone.getObjectTransform();
         }
-        return getVertexNormals(positions, rotations);
+        return getVertexNormals(Arrays.asList(transforms));
     }
 
-    public List<Vector3f> getVertexPositions(List<Vector3f> bonePositions, List<Quat4f> boneRotations) {
+    public List<Vector3f> getVertexPositions(List<Matrix4f> boneTransforms) {
```

Comment:
Is there a process for auto-generating documentation? Would documenting these methods potentially be useful for new developers to become familiar with the skeletal mesh API?

Reply:
If the documentation is autogenerated then it is likely not providing any information of value (since it would all be derived from the method name/parameter names/etc.

I'll chuck some javadoc on these methods. I'm not sure they would be used much - primarily they exist for the rendering system and I'm not sure there is much reason to otherwise interact with them, maybe custom collision checking against exact mesh position I guess.

Reply:
OK, I just noticed they were public methods, so thought they might be used by a developer somehow.